bundles/org.openhab.binding.mycroft/src/main/java/org/openhab/binding/mycroft/internal/channels/VolumeChannel.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mycroft.internal.channels;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.mycroft.internal.MycroftBindingConstants;
+import org.openhab.binding.mycroft.internal.MycroftHandler;
+import org.openhab.binding.mycroft.internal.api.MessageType;
+import org.openhab.binding.mycroft.internal.api.dto.BaseMessage;
+import org.openhab.binding.mycroft.internal.api.dto.MessageVolumeDecrease;
+import org.openhab.binding.mycroft.internal.api.dto.MessageVolumeGet;
+import org.openhab.binding.mycroft.internal.api.dto.MessageVolumeGetResponse;
+import org.openhab.binding.mycroft.internal.api.dto.MessageVolumeIncrease;
+import org.openhab.binding.mycroft.internal.api.dto.MessageVolumeSet;
+import org.openhab.core.library.types.IncreaseDecreaseType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+
+/**
+ * The channel responsible for handling the volume of the Mycroft speaker
+ * NOT FUNCTIONAL
```

Comment:
In fact, the volume channel is quite usable right now.
But it is **despite** Mycroft. The volume management on Mycroft side is IMHO still broken (no update on this subject since a loooong time)
See this one for example : https://github.com/MycroftAI/skill-volume/issues/53

I made some workarounds in my last commit. A little bit ugly, but it's now OK.
- I dropped the "duck" channel, which was in fact almost a duplicate of the "mute" channel and did'nt work like I thought, - I also forged a special message to force Mycroft to set volume as the binding wants. It is a workaround for the message bus API inconsistency (I simulate an volume change intent on the message bus instead of a direct volume command)
- I provided an option to mitigate a bug in the way Mycroft restore volume (the binding can now force a specific volume level after an unmute action)

The major issue, that I couldn't overcome, is that the binding don't have a way of asking Mycroft the current volume level.
It can (and does) follow the volume changes during the time it is connected, so it is not as big an issue as it sounds.

Another issue : it's a workaround ! It is subject to Mycroft internal change.

And one interrogation : it seems the Mark I handles sound differently (its volume management is not in the volume skill, but in the enclosure module). As I don't have one, I'm not able to test its behaviour with the messages exchanged in the message bus.

Despite this, and as it is quite usable, I now think we can reactive the volume channel. Are you OK with that ?

Reply:
Yes, absolutely ok with it!